hazelcast/src/main/java/com/hazelcast/internal/monitors/OverloadedConnectionsPlugin.java

```diff
@@ -0,0 +1,229 @@
+import static java.lang.Math.min;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * The PacketSampler is a debug aid that periodically scans the Connection.writeQueues and samples the content.
+ * This helps to figure out what is happening when writeQueues are very large. The PacketSampler is disabled
+ * by default.
+ */
+public class OverloadedConnectionsPlugin extends PerformanceMonitorPlugin {
+    private final SerializationService serializationService;
+    private final ILogger logger;
+    private final Map<String, MutableInteger> occurrenceMap = new HashMap<String, MutableInteger>();
+    private final ArrayList<OutboundFrame> packets = new ArrayList<OutboundFrame>();
+    private final Random random = new Random();
+    private final NumberFormat defaultFormat = NumberFormat.getPercentInstance();
+    private final NodeEngineImpl nodeEngine;
+    private final int periodSeconds;
+    private final int threshold;
+    private final int samples;
+
+    public OverloadedConnectionsPlugin(NodeEngineImpl nodeEngine) {
+        this.nodeEngine = nodeEngine;
+        this.serializationService = nodeEngine.getSerializationService();
+        this.logger = nodeEngine.getLogger(OverloadedConnectionsPlugin.class);
+        this.defaultFormat.setMinimumFractionDigits(3);
+
+        GroupProperties groupProperties = nodeEngine.getGroupProperties();
+        this.periodSeconds = groupProperties.getInteger(PERFORMANCE_MONITOR_OVERLOADED_CONNECTIONS_PERIOD_SECONDS);
+        this.threshold = groupProperties.getInteger(PERFORMANCE_MONITOR_OVERLOADED_CONNECTIONS_THRESHOLD);
+        this.samples = groupProperties.getInteger(PERFORMANCE_MONITOR_OVERLOADED_CONNECTIONS_SAMPLES);
+    }
+
+    @Override
+    public long getPeriodMillis() {
+        return SECONDS.toMillis(periodSeconds);
+    }
+
+    @Override
+    public void onStart() {
+        logger.info("Plugin:active, period-seconds:" + periodSeconds + " threshold:" + threshold + " samples:" + samples);
+    }
+
+    @Override
+    public void run(BufferedWriter writer) throws IOException {
+        writer.write("OverloadedConnections[");
+
+        Set<TcpIpConnection> connections = getTcpIpConnections();
+        for (TcpIpConnection connection : connections) {
+            clear();
+            scan(writer, connection, false);
+
+            clear();
+            scan(writer, connection, true);
+        }
+
+        writer.write(']');
+    }
+
+    private Set<TcpIpConnection> getTcpIpConnections() {
+        ConnectionManager connectionManager = nodeEngine.getNode().getConnectionManager();
+        if (connectionManager instanceof TcpIpConnectionManager) {
+            return ((TcpIpConnectionManager) connectionManager).getActiveConnections();
+        } else {
+            return Collections.emptySet();
+        }
+    }
+
+    private void scan(BufferedWriter w, TcpIpConnection connection, boolean priority) throws IOException {
```

Comment:
I would prefer `writer` instead of `w`, since that name is used in all other methods of this class.